### ISSUE TITLE
Failed to find valid configuration when passing api string in OpenAIClient.ctr

### ISSUE THREAD

**Issue by GitHubUsername1** [RageAgainstThePixel/com.openai.unity#141]
# Bug Report

## Overview

When i run this code : 

` var api = new OpenAIClient("my api key");`

I'm getting this console ouput : 
```
MissingReferenceException: Failed to find a valid OpenAIConfiguration!
OpenAI.OpenAISettings..ctor (OpenAI.OpenAIConfiguration configuration) (at ./Library/PackageCache/com.openai.unity@7.0.3/Runtime/Authentication/OpenAISettings.cs:34)
OpenAI.OpenAISettings.get_Default () (at ./Library/PackageCache/com.openai.unity@7.0.3/Runtime/Authentication/OpenAISettings.cs:96)
OpenAI.OpenAIClient..ctor (OpenAI.OpenAIAuthentication authentication, OpenAI.OpenAISettings settings) (at ./Library/PackageCache/com.openai.unity@7.0.3/Runtime/OpenAIClient.cs:42)
NewBehaviourScript.Start () (at Assets/New Folder/NewBehaviourScript.cs:12)
```

## To Reproduce
- Create a new project (3D Core). Go through the setup instructions on the main page to install this package i.e. "Via Unity Package Manager and OpenUPM"
- Create a new empty game object.
- Create a new script, attach script to game object.
- Add this code : "var api = new OpenAIClient("my api key goes here");
- Save (shows no compilation errors), Run project (this error appears in the console).

## Expected behavior

No errors (I followed the readme instructions, I hard coded my correct openai api for testing purposes).

## Additional context

I'm quite new to Unity, I apologise in advance if it's something simple (I did search on the error first). I'm guessing that there are more steps involved in being able to use this package than the readme mentions, being a relative noob to Unity I'm not sure what to try. Any help would be greatly appreciated as I'd really like to use the features of this package.

**Comment by GitHubUsername1** [comment]
I found a solution, I tried this suggestion from the readme : 

"Unity Scriptable Object

You can save the key directly into a scriptable object that is located in the Assets/Resources folder.

You can create a new one by using the context menu of the project pane and creating a new OpenAIConfiguration scriptable object."

And the error has gone away. So my fault really for not reading all of the readme. I can see in OpenAIConfiguration.cs there's code for creating a new menu item, i wish i could understand how to put this into action (i tried putting the file into the Editor folder).

**Comment by StephenHodgson** [comment]
I will double check it and make sure that it still works correctly.

**Comment by GitHubUsername1** [comment]
Thanks. Just want to point out something on the readme too please, a particular piece of code on creating an Assistant :

Create Assistant

Create an assistant with a model and instructions.
var api = new OpenAIClient();
var request = new CreateAssistantRequest("gpt-3.5-turbo-1106");
var assistant = await OpenAIClient.AssistantsEndpoint.CreateAssistantAsync(request);The 3rd code line, this wouldn't work for me until I changed it to :
 var assistant = await api.AssistantsEndpoint.CreateAssistantAsync(request);

    On Sunday, December 10, 2023 at 03:32:00 PM GMT, Stephen Hodgson ***@***.***> wrote:  
 
 


I will double check it and make sure that it still works correctly.

**Comment by StephenHodgson** [comment]
Yeah that's my bad for copying and pasting it straight out of the unit test.

**Comment by StephenHodgson** [comment]
Fixed the code snippet examples

**Comment by StephenHodgson** [comment]
This was indeed a recent bug regression. Fixed in 7.0.5
